src/lib.rs

```diff
@@ -1671,6 +1671,52 @@ pub trait Itertools: Iterator {
         adaptors::tuple_combinations(self)
     }
 
+    /// Return an iterator adaptor that iterates over the combinations of the
+    /// elements from an iterator.
+    ///
+    /// Iterator element can be any array of type `Self::Item`.
```

Comment:
Please use "Iterator element type is `[Self::Item; K]`. The iterator produces a new array per iteration, and clones the iterator elements" (just like for the existing `combinations`).